charts/oidc-webhook-authenticator/charts/runtime/templates/deployment.yaml

```diff
@@ -13,13 +13,17 @@ metadata:
     app.kubernetes.io/instance: {{ .Release.Name }}
     app.kubernetes.io/managed-by: {{ .Release.Service }}
 spec:
+  {{- if .Values.autoscaling.hpa.enabled }}
+  replicas: {{ .Values.autoscaling.hpa.minReplicas }}
+  {{- else }}
   replicas: {{ .Values.replicaCount }}
+  {{- end }}
```

Comment:
It is better to not change these lines here, because you might want to enable the HPA with min=1, max=2 and apply the deployment with 2 replicas and this cannot be achieved only with `.Values.autoscaling.hpa.minReplicas`

Reply:
In that case HPA will scale the deployment to 1 replica again. What will be the point to install the chart with `replicas: 2` and then wait for the HPA to scale it down?

Reply:
There might be scenarios where the installer reads the current replicas and sets `.Values.replicaCount` with the same value, e.g. https://github.com/gardener/gardener-extension-shoot-oidc-service can do this.

Reply:
I can now see your point. If HPA scales the deployment to some count we want to persist it during deployment.